lib/spack/spack/build_systems/oneapi.py

```diff
@@ -111,3 +117,36 @@ def libs(self):
         lib_path = join_path(self.component_path, 'lib', 'intel64')
         lib_path = lib_path if isdir(lib_path) else dirname(lib_path)
         return find_libraries('*', root=lib_path, shared=True, recursive=True)
+
+
+class IntelOneApiStaticLibraryList(object):
+    """Provides ld_flags when static linking is needed
+
+    oneapi puts static and dynamic libraries in the same directory, so
+    -l will default to finding the dynamic library. Use absolute
+    paths, as recommended by oneapi documentation.
+
+    Allow both static and dynamic libraries to be supplied by the
+    package.
+    """
+
+    def __init__(self, static_libs, dynamic_libs):
+        self.static_libs = static_libs
+        self.dynamic_libs = dynamic_libs
+
+    @property
+    def directories(self):
+        return self.dynamic_libs.directories
+
+    @property
+    def search_flags(self):
+        return self.dynamic_libs.search_flags
+
+    @property
+    def link_flags(self):
+        return '-Wl,--start-group {0} -Wl,--end-group {1}'.format(
+            ' '.join(self.static_libs.libraries), self.dynamic_libs.link_flags)
```

Comment:
(question: no change needed) generally `--start-group ... --end-group` are for cyclic dependencies: is that the case here for the static libs?

Reply:
The mkl documentation doesn't explain why it is needed, but I believe the reason is cyclic dependencies.